### DISCUSSION TITLE
Am I able to set per site notifications?

### DISCUSSION THREAD

**Discussion by dillfrescott** [upptime/upptime#669] [Q&A] [answered]
Such as having smtp alerts be sent out when `google.com` is down but not when `wikipedia.org` is down?

**Comment by AnandChowdhary** [comment, marked as answer]
Definitely a good idea to add! Currently there's no way to do this unfortunately, the only hack is to create multiple Upptime repositories. :(

**Reply by drzraf** [reply]
An idea would be to pass `site.name` to `sendNotification` in  `src/update.ts`
And, in `notifme`, rely on a JSON-encoded variable, eg `NOTIFICATION_POLICIES` to select/limit/map between site names and channels.

Does it sound a good approach? Would you merge a PR going that way?

**Reply by drzraf** [reply]
@AnandChowdhary ?

**Reply by AnandChowdhary** [reply]
@drzraf sounds good, happy to review a PR! Something like this may also work instead of JSON:

```yaml
sites:
  - name: Google
    url: https://www.google.com
    notifications: [discord, slack]
```

And if the `notifications` key is not provided, we assume all of them, so you can use `notifications: []` to disable all notifications for a particular site. What say?

**Reply by drzraf** [reply]
But notifications should rather references names rather than method so that, eg `google.com` notifies `["telegram@us"]` but `google.fr` notifies `["slack-all", "telegram-fr-group"]`

**Reply by AnandChowdhary** [reply]
That makes sense, yes! But we use repository secrets for notifications, so there's no way to configure two separate Slack notifications, right? Or am I mistaken? :)

**Reply by drzraf** [reply]
Indeed. Only one "channel" per method currently.
AFAICT, there are two options:
- Either adding a separate mapping system, like a more flexible JSON variable secret replacing individual secret.
- Or using `SLACK_<NAME>_*`-like variable. More complex, less powerful, less user friendly and based on the possibility of iterating of the environment (not always practicable)

**Comment by drzraf** [comment]
I believe it'd be a very useful feature:
- Sites A and B downtime could be not only assigned to sysadmin X but notified to him.
- On the other side C and D notifications would only apply to person Y notified on a different phone number.
